content/src/view.js

```diff
@@ -2208,44 +2208,15 @@ function changeEditorText(paneState, text) {
   if (text.length && text.charAt(text.length - 1) != '\n') {
     text += '\n';
   }
-  paneState.changeHandler.suppressChange = true;
-  var saved = {}, editor = paneState.editor, session = editor.session;
-  saved.selection = session.selection.toJSON()
-  saved.atend = session.selection.getCursor().row >= session.getLength() - 1;
-  saved.folds = session.getAllFolds().map(function(fold) {
```

Comment:
Yup, so the call to set text is used via external API.  The idea is that a tutorial website could set the text on a pencilcode program as part of a tutorial.

All the care with selections etc means that, if a student already has some code, setting the text will modify the code but keep the selection in the same place.